src/engine/pieces/pawn.ts

```diff
@@ -1,13 +1,42 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
 
 export default class Pawn extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let availableMoves = new Array<Square>();
+        let finalAvailableMoves = [];
+        let currentPosition = board.findPiece(this);
+        let oldRow = currentPosition.row;
+        let oldCol = currentPosition.col;
+
+        if (this.player == 0) {
```

Comment:
do we not want to be using the existing player enum here? e.g. `Player.WHITE`

Reply:
Fixed